src/filters/truncate.js

```diff
@@ -3,7 +3,7 @@
 angular.module('openshiftCommonUI')
   .filter('truncate', function() {
     return function(str, charLimit, useWordBoundary, newlineLimit) {
-      if (!str) {
+      if (typeof str !== 'string' || typeof charLimit !== 'number') {
```

Comment:
ah didn't account for an either both, or just one, scenario on ```charLimit``` and ```newlineLimit```. updating the condition